lib/kb/models/pet_contract.rb

```diff
@@ -67,5 +67,9 @@ def product
     def pet
       @pet ||= Pet.find(pet_key)
     end
+
+    def hubspot_id
+      @hubspot_id ||= KB::Hubspot.relationship('policies', key).hubspot_id
+    end
```

Comment:
Are you ok with this attribute accessor throwing an exception ?
The KB::Hubspot.relationship('policies', key) can potentially throw a KB::ResourceNotFound exception.

Reply:
Finally we've added the`KB::ResourceNotFound` on `HubspotRelationship` model. But this `hubspot_id` method will return a nil in case of error.